lib/semian/adapter.rb

```diff
@@ -0,0 +1,34 @@
+module Semian
+  module Adapter
+    def semian_identifier
+      raise NotImplementedError.new("Semian adapters must implement a `semian_identifier` method")
+    end
+
+    private
+
+    def acquire_semian_resource(scope:)
+      return yield if @resource_acquired # Prevent a single adapter from acquiring the resource twice by nesting calls
```

Comment:
Redis basically does:

```ruby
def get
  ensure_connected do
    actuall_get
  end
end
```

Which ended up doing a 2 nested acquires. I believe we better prevent that.